definitions/npm/styled-components_v2.x.x/flow_v0.53.x-/styled-components_v2.x.x.js

```diff
@@ -11,7 +11,8 @@ type $npm$styledComponents$ReactComponentFunctional<Props, DefaultProps = *> = {
   defaultProps: DefaultProps,
 }
 
-type $npm$styledComponents$ReactComponentFunctionalUndefinedDefaultProps<Props> = { (Props): React$Node }
+type $npm$styledComponents$ReactComponentFunctionalUndefinedDefaultProps<Props> =
```

Comment:
If it's `React$StatelessFunctionalComponent<Props>`, why not just use `React$StatelessFunctionalComponent<Props>`?

Reply:
Only for clarity of the definitions...
```js
type $npm$styledComponents$ReactComponentUnionWithDefaultProps<Props, DefaultProps> =
  | $npm$styledComponents$ReactComponentFunctional<Props, DefaultProps>
  | $npm$styledComponents$ReactComponentFunctionalUndefinedDefaultProps<Props>
  | $npm$styledComponents$ReactComponentClass<Props, DefaultProps>
  | $npm$styledComponents$ReactComponentClassUndefinedDefaultProps<Props>
```

Reply:
These global types are already not fun to work with, so trying to make it somewhat easier to understand...

Reply:
I thought about using underscores to make it easier on the eyes, but it's a bit unconventional...
```js
type $npm$styledComponents$ReactComponent__Union__WithDefaultProps<Props, DefaultProps> =
  | $npm$styledComponents$ReactComponent__Functional<Props, DefaultProps>
  | $npm$styledComponents$ReactComponent__Functional__UndefinedDefaultProps<Props>
  | $npm$styledComponents$ReactComponent__Class<Props, DefaultProps>
  | $npm$styledComponents$ReactComponent__Class__UndefinedDefaultProps<Props>
```

Reply:
I fail to see how `$npm$styledComponents$ReactComponentFunctionalUndefinedDefaultProps<Props>` is clearer than `React$StatelessFunctionalComponent<Props>`, but it's your call.